src/js/core/textual_game_state.tsx

```diff
@@ -1,40 +1,76 @@
 import { HUDModalDialogs } from "../game/hud/parts/modal_dialogs";
 import { GameState } from "./game_state";
-import { T } from "../translations";
 
 /**
  * Baseclass for all game states which are structured similary: A header with back button + some
  * scrollable content.
  */
-export class TextualGameState extends GameState {
-    ///// INTERFACE ////
+export abstract class TextualGameState extends GameState {
+    private backToStateId: string | null = null;
+    private backToStatePayload: {} | null = null;
+
+    protected headerElement: HTMLElement;
+    protected containerElement: HTMLElement;
+    protected dialogs: HUDModalDialogs;
 
     /**
      * Should return the states inner html. If not overriden, will create a scrollable container
      * with the content of getMainContentHTML()
```

Comment:
comment should be moved to `getContentLayout()` probably

Reply:
Done, kind of.